signac/job.py

```diff
@@ -406,6 +414,33 @@ def update_statepoint(self, update, overwrite=False):
         statepoint.update(update)
         self.statepoint = statepoint
 
+    @property
+    def cached_statepoint(self):
```

Comment:
Must we add a new public API in order to provide the performance benefits of this PR? I am unsure if we should permit users to call this, or if it should be only leveraged internally as a private property `job._cached_statepoint`.

Reply:
Thanks for the explanation, that is helpful.

I would be open to changing `statepoint` semantics to be read-only in a future major version. We had discussed this at one point as a possibility for signac 2. Let's file an issue for that proposal.

Reply:
Noted in #983.